rita/src/rita_exit/db_client/mod.rs

```diff
@@ -110,121 +108,243 @@ fn verify_identity(details: &ExitRegistrationDetails, request_ip: &IpAddr) -> Re
+    }
+}
+
+fn add_dummy(conn: &SqliteConnection) -> Result<(), Error> {
+    use self::schema::clients::dsl::*;
+
+    let dummy = models::Client {
+        mesh_ip: "0.0.0.0".to_string(),
+        wg_pubkey: "".to_string(),
+        wg_port: "".to_string(),
+        luci_pass: "".to_string(),
+        internal_ip: SETTING.get_exit_network().exit_start_ip.to_string(),
+        email: "".to_string(),
+        zip: "".to_string(),
+        country: "".to_string(),
+    };
+
+    match diesel::insert_into(clients).values(&dummy).execute(&*conn) {
+        Err(e) => warn!("got error inserting dummy: {}", e),
+        _ => {}
+    }
+    Ok(())
+}
+
+fn incr_dummy(conn: &SqliteConnection) -> Result<IpAddr, Error> {
+    use self::schema::clients::dsl::*;
+
+    add_dummy(&conn)?;
+    let dummy: models::Client = clients
+        .filter(mesh_ip.eq("0.0.0.0"))
+        .load::<models::Client>(&*conn)
+        .expect("failed loading dummy")[0]
+        .clone();
+
+    trace!("dummy: {:?}", dummy);
+
+    let new_ip = increment(dummy.internal_ip.parse()?)?;
+
+    diesel::update(clients.filter(mesh_ip.eq("0.0.0.0")))
+        .set(internal_ip.eq(&new_ip.to_string()))
+        .execute(&*conn)?;
+
+    Ok(new_ip)
+}
+
+fn update_client(client: &ExitClientIdentity, conn: &SqliteConnection) -> Result<(), Error> {
+    use self::schema::clients::dsl::*;
+    diesel::update(clients.find(&client.global.mesh_ip.to_string()))
+        .set(wg_port.eq(&client.wg_port.to_string()))
+        .execute(&*conn)?;
+
+    diesel::update(clients.find(&client.global.mesh_ip.to_string()))
+        .set(wg_pubkey.eq(&client.global.wg_public_key.clone()))
+        .execute(&*conn)?;
+
+    diesel::update(clients.find(&client.global.mesh_ip.to_string()))
+        .set(email.eq(&client.reg_details.email.clone().unwrap()))
+        .execute(&*conn)?;
+
+    diesel::update(clients.find(&client.global.mesh_ip.to_string()))
+        .set(zip.eq(&client.reg_details.zip_code.clone().unwrap()))
+        .execute(&*conn)?;
+
+    Ok(())
+}
+
+fn client_exists(ip: &IpAddr, conn: &SqliteConnection) -> Result<bool, Error> {
+    use self::schema::clients::dsl::*;
+    Ok(select(exists(clients.filter(mesh_ip.eq(ip.to_string())))).get_result(&*conn)?)
+}
+
+fn client_to_db_client(
+    client: ExitClientIdentity,
+    new_ip: IpAddr,
+    country: String,
+) -> models::Client {
+    models::Client {
+        luci_pass: "".into(),
+        wg_port: client.wg_port.to_string(),
+        mesh_ip: client.global.mesh_ip.to_string(),
+        wg_pubkey: client.global.wg_public_key.clone(),
+        internal_ip: new_ip.to_string(),
+        email: client.reg_details.email.clone().unwrap_or("".to_string()),
+        zip: client
+            .reg_details
+            .zip_code
+            .clone()
+            .unwrap_or("".to_string()),
+        country,
+    }
+}
+
+fn email_ver_done(_mesh_ip: &IpAddr, _conn: &SqliteConnection) -> Result<bool, Error> {
```

Comment:
does this need to be finished, also where is sending the email itself?

Reply:
ok so we still need integration with fastmail / some mail service

Reply:
yeah, planning to do that in a new PR as it is pretty self contained (and this one is big enough as is)